src/noPipeImpureRule.ts

```diff
@@ -31,13 +32,13 @@ export class Rule extends AbstractRule {
 }
 
 export class ClassMetadataWalker extends NgWalker {
-  protected visitNgPipe(controller: ClassDeclaration, decorator: Decorator): void {
-    this.validatePipe(controller, decorator);
-    super.visitNgPipe(controller, decorator);
+  protected visitNgPipe(metadata: PipeMetadata): void {
+    this.validatePipe(metadata);
+    super.visitNgPipe(metadata);
   }
 
-  private validatePipe(controller: ClassDeclaration, decorator: Decorator): void {
-    const pureExpression = getDecoratorPropertyInitializer(decorator, 'pure');
+  private validatePipe(metadata: PipeMetadata): void {
+    const pureExpression = getDecoratorPropertyInitializer(metadata.decorator, 'pure');
```

Comment:
Now, I guess we can just `metadata.pure`, right? Do you prefer to do this kind of "refactor" in a new PR?

Reply:
It's done